src/modules/api/user/link/link.controller.ts

```diff
@@ -46,18 +36,21 @@ export class LinkController {
   ): Promise<IResult<LinkDTO>> {
     try {
       const link = await this.linkService.create(linkData, userId);
-      const linkDTO = {
-        _id: link._id,
-        name: link.name,
-        customUrl: link.customUrl,
-        redirectsTo: link.redirectsTo,
-        status: link.status,
-        inPool: link.inPool,
-      } as LinkDTO;
-
+      const linkDTO = new LinkDTO(link);
       return { status: 'OK', result: linkDTO };
     } catch (error) {
       return { status: 'NOT OK', error: error.message };
     }
   }
+
+  @Delete(':id')
+  async deleteLink(@Param('id') linkId: string): Promise<IResult<LinkDTO>> {
```

Comment:
naming it just 'delete' would be better i guess. Or is this done since delete might be a keyword elsewhere ?

Reply:
delete is not used anywhere else. thought it would be more readable.

Reply:
Got it. Either we do that for everything, like createLink, deleteLink, or stick to create,delete everywhere. I think it would be more homogeneous that way.

Reply:
Service side I think it's better to go with just create, delete and edit since we do `linkService.create, userService.create` and let's have createLink for Controllers side.